src/base/http/requestparser.cpp

```diff
@@ -210,7 +211,10 @@ bool RequestParser::parseRequestLine(const QString &line)
     const int sepPos = url.indexOf('?');
     const QByteArray pathComponent = ((sepPos == -1) ? url : midView(url, 0, sepPos));
 
-    m_request.path = QString::fromUtf8(QByteArray::fromPercentEncoding(pathComponent));
+    // Convert all '\' to '/'
+    // This helps the server down the line to guard against path traversal using '.' or '..'
+    m_request.path = QString::fromUtf8(QByteArray::fromPercentEncoding(pathComponent))
+                         .replace(u"\\"_qs, u"/"_qs);
```

Comment:
I would another overload: https://doc.qt.io/qt-6/qstring.html#replace-3
The last time I checked its source, it was simpler (and faster I presume).